### PR TITLE
bug 1170204 - further adjust ES heap sizes to fit current instance types

### PR DESCRIPTION
**Pull request by rhelmer**
r? @phrawzty

### CODE REVIEW

**Comment by phrawzty** [comment]
Within the context of ES, heap sizes are governed by [two factors](https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html): Lucene and Java.

Lucene operates out of the heap (it's functionally filesystem operations), and therefore has separate memory requirements. As a general rule, one should never allocate more than 50% of the total system RAM to the ES Java heap - the other half is left to the kernel to be used for the system, which includes Lucene.

Java catches a bad case of *the dumb* when it has more than 32GB of memory to deal with. Allocating more than that to the heap becomes a question of diminishing returns, and can even have detrimental effects for smaller allocation sizes; unless one is allocating *hundreds* of gigs it's not even worth dealing with the issue.

In summary, these numbers should not exceed 32GB, and must not exceed 50% of total system.

**Comment by phrawzty** [comment]
`r+`

As an aside, there is a potential argument to be made for violating the "50%-rule" on the *master* nodes since they aren't actually doing any heavy lifting with Lucene. Something to think about for the future, perhaps.
